pyaerocom/plugins/harp/reader.py

```diff
@@ -0,0 +1,227 @@
+
+    @property
+    def PROVIDES_VARIABLES(self) -> List[str]:
+        return list(self.VAR_MAPPING.keys())
+
+    def _get_station_data(self, data: pd.DataFrame) -> StationData:
+        sd = StationData()
+
+        return sd
+
+    def _get_stationname(self, filename: str) -> str | None:
+        station_pattern = "MEP-surface-rd-(.*A)-.*.nc"
+        try:
+            stationname = re.search(station_pattern, filename).group(1)
+            return stationname
+
+        except:
+            print(f"Skipping file {filename}")
+            logger.debug(f"Skipping file {filename}")
+            return None
+
+    def _get_time(self, data: xr.Dataset) -> List[np.datetime64]:
+
+        start = data["datetime_start"].data
+        end = data["datetime_stop"].data
+
+        return list(start)
+
+    def read_file(self, filename: str, vars_to_retrieve: List[str]) -> xr.Dataset:
+        """Reads data for a single year for one component"""
+        return xr.open_dataset(filename)
+
+    def read(
+        self, vars_to_retrieve=None, files=None, first_file=None, last_file=None, metadatafile=None
+    ):
+        if vars_to_retrieve is None:
+            vars_to_retrieve = self.PROVIDES_VARIABLES
+
+        for var in vars_to_retrieve:
+            if var not in self.PROVIDES_VARIABLES:
+                raise ValueError(f"The variable {var} is not supported")
+
+        stations: List[Station] = []
+
+        for stationname in tqdm(self.STATIONS):
+            f = self.STATIONS[stationname][0]
+
+            data = self.read_file(f, vars_to_retrieve)
+            lat = float(data["latitude"])
+            lon = float(data["longitude"])
+            alt = float(data["altitude"])
+            station = Station(stationname, lat, lon, alt)
+
+            for f in self.STATIONS[stationname]:
+
+                for s in vars_to_retrieve:
+                    try:
+                        measurements = data[self.VAR_MAPPING[s]].data
+                        unit = data[self.VAR_MAPPING[s]].units
+                    except:
+                        print(f"Could not find {s} in {f}. Skipping file")
+                        logger.info(f"Could not find {s} in {f}. Skipping file")
+                        continue
+
+                    time = self._get_time(data)
+                    for t, m in zip(time, measurements):
+                        station.add_measurement(s, unit, m, t)
+
+            stations.append(
+                station.to_stationdata(self.DATA_ID, self.DATASET_NAME, self.STATIONS[stationname])
+            )
+
+        return UngriddedData.from_station_data(
+            stations
+        )  # , add_meta_keys=list(set(metadata_headers)))
+
+
+if __name__ == "__main__":
+    FILE_DIR = "/lustre/storeA/project/aerocom/aerocom1/AEROCOM_OBSDATA/CHINA_MP_NRT/HARP_EXAMPLE_DATA/download/JJA-2022/HARP2/nobackup/users/tsikerde/China_grounddata/HARP/"
+    reader = ReadHARP(data_dir=FILE_DIR)
+    data = reader.read()
+
+    breakpoint()
```

Comment:
:)

Reply:
One can never get enough breakpoints:) It's there so I can inspect the results. The whole __name__ will of course be removed when everything is done

Reply:
you can stage part of the file, so the code that is not meant to be part of the repository (like this `if __name__ == "__main__":` clause) is never commited.